core/src/main/java/hudson/Functions.java

```diff
@@ -1907,20 +1907,14 @@ public static String joinPath(String... components) {
     }
 
     /**
-     * Computes the link to the console for the run for the specified executable, taking {@link ConsoleUrlProvider} into account.
-     * @param executable the executable (normally a {@link Run})
-     * @return the absolute URL for accessing the build console for the executable, or null if there is no build associated with the executable
+     * Computes the link to the console for the run for the specified object, taking {@link ConsoleUrlProvider} into account.
+     * @param withConsoleUrl the object to compute a console url for (can be {@link Run}, a {@code PlaceholderExecutable}...)
```

Comment:
After the last rework it isn't an object any longer, maybe also adjust the javadoc here

Reply:
Well, a “model object”. Seems OK to me.